portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -45,28 +45,27 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
    PreparedQuery results = datastore.prepare(query);
    
-   List<Task> task = new ArrayList<Task>();
+   List<Task> taskList = new ArrayList<Task>();
    for (Entity entity : results.asIterable()) {
       long id = entity.getKey().getId();
+      String email = (String) entity.getProperty("email");
       String text = (String) entity.getProperty("text");
       
-      Task task2 = new Task(id, text);
-      task.add(task2);
+      Task task = new Task(id, text, email);
+      taskList.add(task);
     }
 
     Gson gson = new Gson();
     response.setContentType("application/json");
-    response.getWriter().println(gson.toJson(task));
+    response.getWriter().println(gson.toJson(taskList));
  }
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the input from the form.
-    String text = getParameter(request, "text-input", "");
     
+    String text = getParameter(request, "text-input", "");
     
-
-
     UserService userService = UserServiceFactory.getUserService();
-    
+    String email = userService.getCurrentUser().getEmail();
     Entity taskEntity = new Entity("Task");
     taskEntity.setProperty("text", text);
```

Comment:
You're getting the email from the UserService but it looks like it's not getting stored on the Task